### PR TITLE
[fix]: Added error checks to `mark_internal_external`

### PR DESCRIPTION
**Pull request by BenTaylor-TfN**
# Changes

Added 2 error checks to `mark_internal_external`.
- If the origin and destination columns have different dtypes.
- If no internal zones are found in either the origin or destination columns.

# Tasks

- [x] Have unittests been added (testing individual functions and their edge cases)
- [x] Has documentation (including user guide) been updated
- [x] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [x] Code linting, tests and other workflows pass

----
📚 Documentation preview 📚: https://caftoolkit--216.org.readthedocs.build/en/216/

### CODE REVIEW

File: src/caf/toolkit/pandas_utils/matrices.py [hunk 1]
```diff
@@ -580,16 +580,36 @@ def mark_internal_external(
...
 
     marked_matrix = matrix.copy()
     origin_in = marked_matrix[origin_col].isin(internal_zones)
     dest_in = marked_matrix[destination_col].isin(internal_zones)
 
+    if not origin_in.any() and not dest_in.any():
+        raise ValueError(
+            "No internal zones found in either origin or destination columns. "
+            "Are the datatypes the same?"
+        )
```
Comment (MattBuckley-TfN) [inline src/caf/toolkit/pandas_utils/matrices.py:612]: Might be worth adding a warning if one is missing but not the other
```suggestion
        )
    if (not origin_in.any()) ^ (not dest_in.any()):
        if origin_in.any():
            columns = ("origin", "destination")
        else:
            columns = ("destination", "origin")
        warnings.warn(
            f"Internal zones only found in {columns[0]}"
            f" column, not in {columns[1]} column",
            stacklevel=2,
        )
```

**Review by MattBuckley-TfN — COMMENTED** [review]
One suggestion to add a warning but happy to accept if you don't think it's needed.

**Review by MattBuckley-TfN — APPROVED** [review]
LGTM
